src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/nd4jutils/Nd4jIOUtils.java

```diff
@@ -0,0 +1,161 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.nd4jutils;
+
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.tsv.DataLine;
+import org.broadinstitute.hellbender.utils.tsv.TableColumnCollection;
+import org.broadinstitute.hellbender.utils.tsv.TableReader;
+import org.broadinstitute.hellbender.utils.tsv.TableWriter;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.IntStream;
+
+/**
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class Nd4jIOUtils {
+
+    /**
+     * Write NDArray to a binary dump
+     *
+     * @param arr an instance of {@link INDArray}
+     * @param outputFile the output path
+     */
+    public static void writeNDArrayToBinaryDumpFile(@Nonnull final INDArray arr,
+                                                    @Nonnull final File outputFile) {
+        try (final DataOutputStream dos = new DataOutputStream(new FileOutputStream(outputFile))) {
+            Nd4j.write(arr, dos);
+        } catch (final IOException ex) {
+            throw new UserException.CouldNotCreateOutputFile(outputFile, "Could not write the NDArray");
+        }
+    }
+
+    /**
+     * Read NDArray from a binary dump
+     *
+     * @param inputFile the input Nd4j binary dump file
+     * @return an {@link INDArray}
+     */
+    public static INDArray readNDArrayFromBinaryDumpFile(@Nonnull final File inputFile) {
+        try (final DataInputStream dis = new DataInputStream(new FileInputStream(inputFile))) {
+            return Nd4j.read(dis);
+        } catch (final IOException ex) {
+            throw new UserException.CouldNotReadInputFile(inputFile, "Could not read the input file");
+        }
+    }
+
+    /**
+     * Write NDArray to a tab-separated file
+     *
+     * @param arr an instance of {@link INDArray}
```

Comment:
Extended the javadocs significantly, especially after the new stuff introduced by #4891f92.